nibabel/streamlines/__init__.py

```diff
@@ -0,0 +1,132 @@
+import os
+import warnings
+from ..externals.six import string_types
+
+from .header import Field
+from .array_sequence import ArraySequence
+from .tractogram import Tractogram, LazyTractogram
+from .tractogram_file import ExtensionWarning
+
+from .trk import TrkFile
+
+# List of all supported formats
+FORMATS = {".trk": TrkFile}
+
+
+def is_supported(fileobj):
+    """ Checks if the file-like object if supported by NiBabel.
+
+    Parameters
+    ----------
+    fileobj : string or file-like object
+        If string, a filename; otherwise an open file-like object pointing
+        to a streamlines file (and ready to read from the beginning of the
+        header)
+
+    Returns
+    -------
+    is_supported : boolean
+    """
+    return detect_format(fileobj) is not None
+
+
+def detect_format(fileobj):
+    """ Returns the StreamlinesFile object guessed from the file-like object.
+
+    Parameters
+    ----------
+    fileobj : string or file-like object
+        If string, a filename; otherwise an open file-like object pointing
+        to a tractogram file (and ready to read from the beginning of the
+        header)
+
+    Returns
+    -------
+    tractogram_file : :class:`TractogramFile` class
+        The class type guessed from the content of `fileobj`.
+    """
+    for format in FORMATS.values():
+        try:
+            if format.is_correct_format(fileobj):
+                return format
+        except IOError:
+            pass
+
+    if isinstance(fileobj, string_types):
+        _, ext = os.path.splitext(fileobj)
+        return FORMATS.get(ext.lower())
```

Comment:
What you said is correct. Note also that this part of the function is mainly used when saving a `Tractogram` object (rather than a `TractogramFile` object) using `nib.streamlines.save` since we don't have any other way of guessing the file format.